### PR TITLE
[15168] Notify changes in bulk on RTPS readers

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

This PR implements calls to the new `on_data_available` callback of `rtps::ReaderListener`

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- *N/A* Documentation builds and tests pass locally. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by JLBuenoLopez-eProsima — DISMISSED** [review]
LGTM with CI!

I assume it is not easy to prepare a deterministic test that ensures that some times a single message is notified and other times changes in a bulk are notified.

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

**Comment by MiguelCompany** [comment]
@richiprosima Please test windows and test linux also

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiware Would you mind reviewing this?

File: src/cpp/rtps/common/ChangeComparison.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#ifndef _RTPS_COMMON_CHANGECOMPARISON_HPP
+#define _RTPS_COMMON_CHANGECOMPARISON_HPP
+
+#include <fastdds/rtps/common/CacheChange.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+
+inline bool history_order_cmp(
```
Comment (richiware) [inline src/cpp/rtps/common/ChangeComparison.hpp:28]: I would write specific tests for this kind of sorting.

**Review by richiware — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/common/ChangeComparison.hpp (hunk 1 above)
Comment (MiguelCompany) [reply src/cpp/rtps/common/ChangeComparison.hpp:28]: Test added in 261204f

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

File: test/unittest/rtps/history/ReaderHistoryTests.cpp [hunk 2]
```diff
@@ -171,6 +176,64 @@ TEST_F(ReaderHistoryTests, get_change)
...
+        GUID_t writer_guid = GUID_t(GuidPrefix_t::unknown(), i);
+
+        for (uint32_t j = 1; j <= changes_per_writer; j++)
+        {
+            CacheChange_t* ch = new CacheChange_t(0);
+            ch->writerGUID = writer_guid;
+            ch->sequenceNumber = SequenceNumber_t(0, j);
+            ch->sourceTimestamp = rtps::Time_t(0, ts);
+
+            changes_list.push_back(ch);
```
Comment (richiware) [inline test/unittest/rtps/history/ReaderHistoryTests.cpp:198]: It should be `changes` instead of `changes_list`, shouldn't be?

**Review by richiware — CHANGES_REQUESTED** [review]

File: test/unittest/rtps/history/ReaderHistoryTests.cpp (hunk 2 above)
Comment (MiguelCompany) [reply test/unittest/rtps/history/ReaderHistoryTests.cpp]: Test changed. See dc35cef

**Review by richiware — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test windows

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

**Pull request merged**